transformer_engine/common/fused_attn/fused_attn_fp16_bf16_max_seqlen_512.cu

```diff
@@ -758,16 +776,17 @@ void fused_attn_max_512_fwd_impl(int64_t b, int64_t h, int64_t s_q, int64_t s_kv
         data_ptrs.insert(std::pair<uint64_t, void *>(K_SEQLEN_ID, devActualSeqlenK));
         data_ptrs.insert(std::pair<uint64_t, void *>(MASK_VAL_ID, &negInfinity));
 
+        __half half_cast_scaling_factor{scaling_factor};
+        __nv_bfloat16 bfloat_cast_scaling_factor{scaling_factor};
+
         if (tensorType == CUDNN_DATA_FLOAT) {
             data_ptrs.insert(std::pair<uint64_t, void *>(S_CONST_ID, &scaling_factor));
         } else if (tensorType == CUDNN_DATA_HALF) {
-            __half cast_scaling_factor{scaling_factor};
-            data_ptrs.insert(std::pair<uint64_t, void *>(S_CONST_ID, &cast_scaling_factor));
+            data_ptrs.insert(std::pair<uint64_t, void *>(S_CONST_ID, &half_cast_scaling_factor));
         } else if (tensorType == CUDNN_DATA_BFLOAT16) {
-            __nv_bfloat16 cast_scaling_factor{scaling_factor};
-            data_ptrs.insert(std::pair<uint64_t, void *>(S_CONST_ID, &cast_scaling_factor));
+            data_ptrs.insert(std::pair<uint64_t, void *>(S_CONST_ID, &bfloat_cast_scaling_factor));
         } else {
-            std::cerr << "Not supported tensorType." << std::endl;
+            NVTE_ERROR("Not supported tensor type.");
```

Comment:
```suggestion
            NVTE_ERROR("Unsupported tensor type.");
```

Reply:
Fixed in 17a183c73c41c988625ae28c779564d3b5fb2afd